extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -70,8 +70,13 @@ public static function parserUpgradeSpecialFilePathURLs( string &$url ) {
 	 * @return boolean
 	 */
 	public static function onLinkerMakeExternalLink( string &$url, string &$text, bool &$link, array &$attribs ): bool {
-		if ( wfHttpsAllowedForURL( $url ) ) {
-			$url = wfProtocolUrlToRelative( $url );
+		global $wgFandomBaseDomain, $wgWikiaOrgBaseDomain, $wgWikiaBaseDomain;
+
+		$host = explode('.', parse_url( $url, PHP_URL_HOST ));
+		$pattern = '/('. $wgFandomBaseDomain .'|'. $wgWikiaOrgBaseDomain .'|(?:wikia|fandom)-dev.(?:com|pl|us))/';
+
+		if ( preg_match( $pattern, $url ) || ( preg_match('/'. $wgWikiaBaseDomain .'/', $url ) && sizeof( $host ) <= 3 ) ) {
```

Comment:
why do you match the `url` instead of `host`? Won't it break if someone uses `fandom.com` in article name or query params?